src/rules/infinite-reactive-loop.ts

```diff
@@ -0,0 +1,439 @@
+      [ReferenceTracker.ESM]: true,
+      tick: {
+        [ReferenceTracker.CALL]: true,
+      },
+    },
+  })
+  return Array.from(a).map(({ node, path }) => {
+    return {
+      node: node as TSESTree.CallExpression,
+      name: path[path.length - 1],
+    }
+  })
+}
+
+/**
+ * Get usage of `setTimeout`, `setInterval`, `queueMicrotask`
+ */
+function extractTaskReferences(
+  context: RuleContext,
+): { node: TSESTree.CallExpression; name: string }[] {
+  const referenceTracker = new ReferenceTracker(context.getScope())
+  const a = referenceTracker.iterateGlobalReferences({
+    setTimeout: { [ReferenceTracker.CALL]: true },
+    setInterval: { [ReferenceTracker.CALL]: true },
+    queueMicrotask: { [ReferenceTracker.CALL]: true },
+  })
+  return Array.from(a).map(({ node, path }) => {
+    return {
+      node: node as TSESTree.CallExpression,
+      name: path[path.length - 1],
+    }
+  })
+}
+
+/**
+ * If `node` is inside of `maybeAncestorNode`, return true.
+ */
+function isChildNode(
+  maybeAncestorNode: TSESTree.Node | AST.SvelteNode,
+  node: TSESTree.Node,
+): boolean {
+  let parent = node.parent
+  while (parent) {
+    if (parent === maybeAncestorNode) return true
+    parent = parent.parent
+  }
+  return false
+}
+
+/**
+ * Return true if `node` is a function call.
+ */
+function isFunctionCall(node: TSESTree.Node): boolean {
+  if (node.type !== "Identifier") return false
+  const { parent } = node
+  if (parent?.type !== "CallExpression") return false
+  return parent.callee.type === "Identifier" && parent.callee.name === node.name
+}
+
+/**
+ * Return true if `node` is a variable.
+ *
+ * e.g. foo.bar
+ * If node is `foo`, return true.
+ * If node is `bar`, return false.
+ *
+ * e.g. let baz = 1
+ * If node is `baz`, return true.
+ */
+function isVariableNode(node: TSESTree.Identifier): boolean {
+  const { parent } = node
+  if (parent?.type !== "MemberExpression") return true
+  if (
+    parent.type === "MemberExpression" &&
+    parent.object.type !== "Identifier"
+  ) {
+    return false
+  }
+
+  return parent.object.type !== "Identifier"
+    ? false
+    : parent.object.name === node.name
+}
+
+/**
+ * Return true if `node` is a reactive variable.
+ */
+function isReactiveVariableNode(
+  context: RuleContext,
+  node: TSESTree.Node,
+): node is TSESTree.Identifier {
+  if (node.type !== "Identifier") return false
+  if (!isVariableNode(node) || isFunctionCall(node)) return false
+
+  // Variable name starts with `$` means Svelte store.
+  if (node.name.startsWith("$")) return true
+  const scope = getScope(context, node)
+  return scope.references.some((reference) => {
+    const { resolved } = reference
+    if (!resolved || resolved.name !== node.name) return false
+
+    return resolved.defs.some((def) => {
+      return (
+        (def as any).parent?.parent?.type === "SvelteScriptElement" &&
+        def.name.type === "Identifier" &&
+        def.name.name === node.name
+      )
+    })
+  })
```

Comment:
Using `getAllReactiveVariableReferences` as suggested in another comment:

```suggestion
  return getAllReactiveVariableReferences(context).includes(node)
```